content/dev-tools/DestinationAddressInput.tsx

```diff
@@ -0,0 +1,57 @@
+import * as React from 'react';
+import { useTranslate } from '@portal/hooks';
+import { clsx } from 'clsx'
+
+function onDestinationAddressChange(
+    event: React.ChangeEvent<HTMLInputElement>, 
+    setDestinationAddress: React.Dispatch<React.SetStateAction<string>>, 
+    setIsValidDestinationAddress: React.Dispatch<React.SetStateAction<boolean>>
+): void {
+    const newAddress = event.target.value
+    setDestinationAddress(newAddress)
+    // @ts-expect-error - xrpl is guaranteed to be defined by the time this field is changed.
+    setIsValidDestinationAddress(xrpl.isValidAddress(newAddress))
+}
+
+export interface DestinationAddressInputProps {
+    defaultDestinationAddress: string, 
+    destinationAddress: string, 
+    setDestinationAddress: React.Dispatch<React.SetStateAction<string>>, 
+    isValidDestinationAddress: boolean, 
+    setIsValidDestinationAddress: React.Dispatch<React.SetStateAction<boolean>>
+}
+
+export function DestinationAddressInput(
+    {
+        defaultDestinationAddress, 
+        destinationAddress, 
+        setDestinationAddress, 
+        isValidDestinationAddress, 
+        setIsValidDestinationAddress
+    } : DestinationAddressInputProps
+): React.JSX.Element {
+    const { translate } = useTranslate()
+    
+    return (
+    <div>
+        <div className="form-group">
+            <label htmlFor="destination_address">
+                {translate("Destination Address")}
+            </label>
+            <input type="text" className={clsx("form-control", 
+                // Defaults to not having "is-valid" / "is-invalid" classes
+                (destinationAddress !== defaultDestinationAddress) && (isValidDestinationAddress ? "is-valid" : "is-invalid"))}
+                id="destination_address" 
+                onChange={(event) => onDestinationAddressChange(event, setDestinationAddress, setIsValidDestinationAddress)}
+                aria-describedby="destination_address_help" defaultValue={destinationAddress} />
+            <small id="destination_address_help" className="form-text text-muted">
+                {translate("Send transactions to this XRP Testnet address")}
+            </small>
+        </div>
+        <p className={clsx("devportal-callout caution", !(isValidDestinationAddress && destinationAddress[0] === 'X') && "collapse")}
```

Comment:
If it's an r address this hides as expected because the 1st character is not X, this error is specifically for valid X addresses which are pointing at mainnet instead of testnet, so won't work with this tool. 

Using a different writing of the logic:
Only display if this is a valid X address pointed at mainnet. 

Screenshot of the error in action:
![image](https://github.com/XRPLF/xrpl-dev-portal/assets/19694186/d6238cc8-90cd-4b35-ae1a-9f13b71e951a)